spot_price_monitor/spot_price_monitor.py

```diff
@@ -162,6 +168,37 @@ def update_ondemand_price_metrics(metric, prices, types, zones):
             ).set(prices[region][instance_type])
 
 
+def update_ondemand_prices(on_demand, last_ondemand_update, on_demand_spot_gauge, spot_prices, spot_error):
+    '''Refresh ondemand prices once a day'''
+    if on_demand and last_ondemand_update + 86400 < time.time():
```

Comment:
This function's first positional variable gates its entire functionality, returning 0 if not true. In the spirit of making functions reusable, I'd move this logic outside of it, making the function do what it says when invoked.

```suggestion
    if last_ondemand_update + 86400 < time.time():
```